src/scripts/main.js

```diff
@@ -1,3 +1,226 @@
 'use strict';
 
-// write code here
+const body = document.querySelector('body');
+
+let headers, tbody, rows;
+
+const init = () => {
+  headers = [...body.querySelectorAll('th')];
+  tbody = body.querySelector('tbody');
+  rows = [...tbody.querySelectorAll('tr')];
+};
+
+init();
+
+const currencyFormat = (number) =>
+  '$' + new Intl.NumberFormat('en-US').format(number);
+
+// Add active class to row when clicked
+
+rows.forEach((row) => {
+  row.addEventListener('click', () => {
+    rows.forEach((r) => r.classList.remove('active'));
+    row.classList.toggle('active');
+  });
+});
+
+let ASC = true;
+let currentHeader;
+
+// Sort table by column
+headers.forEach((header, i) => {
+  header.addEventListener('click', (e) => {
+    ASC = !ASC;
+
+    if (currentHeader !== i) {
+      ASC = true;
+    }
+    currentHeader = i;
+
+    rows.sort((a, b) => {
+      const cellA = a.querySelectorAll('td')[i].textContent;
+      const cellB = b.querySelectorAll('td')[i].textContent;
+
+      const checkNumberCellA = strToNumber(cellA);
+      const checkNumberCellB = strToNumber(cellB);
+
+      if (Number(checkNumberCellA)) {
+        return ASC
+          ? checkNumberCellA - checkNumberCellB
+          : checkNumberCellB - checkNumberCellA;
+      }
+
+      return ASC ? cellA.localeCompare(cellB) : cellB.localeCompare(cellA);
+    });
+
+    rows.forEach((row) => tbody.appendChild(row));
+  });
+});
+
+const formFields = {
+  name: {
+    label: 'Name',
+    name: 'name',
+    type: 'text',
+    'data-qa': 'name',
+    validation: {
+      minLength: 4,
+    },
+  },
+  position: {
+    label: 'Position',
+    name: 'position',
+    type: 'text',
+    'data-qa': 'position',
+  },
+  office: {
+    label: 'Office',
+    name: 'office',
+    'data-qa': 'office',
+    options: [
+      'Tokyo',
+      'Singapore',
+      'London',
+      'New York',
+      'Edinburgh',
+      'San Francisco',
+    ],
+  },
+  age: {
+    label: 'Age',
+    name: 'age',
+    type: 'number',
+    'data-qa': 'age',
+    validation: {
+      min: 18,
+      max: 90,
+    },
+  },
+  salary: {
+    label: 'Salary',
+    name: 'salary',
+    type: 'number',
+    'data-qa': 'salary',
+  },
+  startDate: '',
+};
+
+const strToNumber = (n) => Number(n.replace(/[^0-9.-]+/g, ''));
```

Comment:
This function `strToNumber` is used to convert a string to a number. It would be more efficient to use the built-in JavaScript function `parseInt` or `parseFloat` instead. Also, this function is used only once in the code. It would be better to use `parseInt` or `parseFloat` directly in that place.